sources/SimulinkCoder/src/CoderBlockInformation.cpp

```diff
@@ -70,102 +99,122 @@ bool CoderBlockInformation::setIOPortsData(const BlockInformation::IOData& /*ioD
 core::BlockInformation::VectorSize
 CoderBlockInformation::getInputPortWidth(const PortIndex idx) const
 {
-    if (pImpl->inputPortDimensions.find(idx) == pImpl->inputPortDimensions.end()) {
-        bfError << "Failed to get width of signal at index " << idx << ".";
+    if (!pImpl->inputPortAtIndexExists(idx)) {
+        bfError << "This block has no input port at index " << idx;
```

Comment:
The block name is printed? Otherwise "This block" may be misleading.

Reply:
This is related to #16. When Simulink shows an error, it automatically adds an header that includes the name of the block in the model that generated it. From the code generated automatically instead this does not happen.

When the block name will be included as in #16, we can obtain a matching behavior. In this particular case, the output shown in the terminal will be the following (the first line is what we need to add):

```
Error in block /model/subsystem/BlockName:
    This block has no input port at index 1
```